triggers/service/src/test/scala/com/digitalasset/daml/lf/engine/trigger/TriggerServiceTest.scala

```diff
@@ -502,6 +502,45 @@ trait AbstractTriggerServiceTestAuthMiddleware
         _ <- resp.status should equal(expectedError)
       } yield succeed
   }
+
+  it should "forbid a non-authorized party to list triggers" in withTriggerService(Nil) {
+    uri: Uri =>
+      val expectedError = StatusCodes.Forbidden
+      for {
+        resp <- listTriggers(uri, eve)
+        _ <- resp.status should equal(expectedError)
+      } yield succeed
+  }
+
+  it should "forbid a non-authorized party to check the status of a trigger" in withTriggerService(
+    List(dar)) { uri: Uri =>
+    val expectedSuccess = StatusCodes.OK
+    val expectedError = StatusCodes.Forbidden
+    for {
+      resp <- startTrigger(uri, s"$testPkgId:TestTrigger:trigger", alice)
+      _ <- resp.status should equal(expectedSuccess)
+      triggerId <- parseTriggerId(resp)
+      _ = authServer.revokeParty(alice)
+      _ = deleteCookies()
```

Comment:
Why do we need to delete cookies? Even if we have a cookie we still call `/authorize` right? Should we make that fail for cookies corresponding to tokens where the party has been revoked? It’s only a test server so you can argue both ways here and I don’t have particularly strong feelings.

Reply:
As described in the PR description this is meant to emulate two different users accessing the trigger service. So, deleting cookies corresponds to the second user not having the first user's cookies available in the first place. I've added a comment to that effect as well.

The revoking of access to parties happens at the oauth2 test server in this case not at the middleware. So, it controls token issuing, not token validation.

> Even if we have a cookie we still call /authorize right?

No, the middleware's /auth endpoint never calls the oauth2 server's /authorize endpoint, that only happens on the middleware's /login endpoint. If the /auth endpoint detects a valid token in the cookie that grants access to the requested claims then it will just return the access token.

Reply:
> > Even if we have a cookie we still call /authorize right?
> 
> No, the middleware's /auth endpoint never calls the oauth2 server's /authorize endpoint, that only happens on the middleware's /login endpoint. If the /auth endpoint detects a valid token in the cookie that grants access to the requested claims then it will just return the access token.

Sorry I meant the trigger service calls `/auth` on the middleware and that could fail. But your explanation with two users makes sense :+1: